rs/interfaces/src/execution_environment.rs

```diff
@@ -1184,6 +1185,8 @@ pub struct RegistryExecutionSettings {
     pub provisional_whitelist: ProvisionalWhitelist,
     pub chain_key_settings: BTreeMap<MasterPublicKeyId, ChainKeySettings>,
     pub subnet_size: usize,
+    /// This field is not from the registry, but from the most recent block.
+    pub replica_version: ReplicaVersion,
```

Comment:
This is a bit confusing to me. Why are we putting it in the `RegistryExecutionSettings` if it doesn't come from the registry? Maybe instead it would make more sense as a new argument to `execute_round`? Or another option would be to combine the other arguments to `execute_round` that have info on the specific block into a single struct argument.

Then I think you also wouldn't need to pass the `ReplicaVersion` through all the message routing logic to read the registry.

Reply:
Ah I see now where this is coming from after looking at the old PR. I'd say do either:

1. Add it as a new argument to `execute_round` and `execute_subnet_message`.
2. Combine it with `idkg_subnet_public_keys` into a struct which is passed to both `execute_round` and `execute_subnet_message`.

Reply:
I share your concerns. The [previous PR](https://github.com/dfinity/ic/pull/2082#discussion_r1810423351) writes batch.replica_version it into the replicated state (metadata) and uses it from there, but it was questioned whether that is necessary. I believe `RegistryExecutionSettings` was suggested because it's already being passed along. 

But I can instead change the signatures of all methods along the way and introduce a new, conceptually clean argument for this data.

Reply:
Here is another draft, PTAL:
https://github.com/dfinity/ic/pull/2248